app/app/helpers/view_helper.rb

```diff
@@ -2,11 +2,11 @@ module ViewHelper
   DATE_FORMAT = "%B %d, %Y"
 
   def format_active_locale(locale_string)
-    link_classes = "usa-nav__link"
-    if locale_string.to_sym == I18n.locale
-      link_classes = "#{link_classes} usa-current"
-    end
-    link_to t("shared.languages.#{locale_string}"), root_path(locale: locale_string), class: link_classes
+    # using request.fullpath to include query parameters so that
+    # switching languages doesn't lose the query parameters
+    current_path = request.fullpath.sub(/\A\/#{I18n.locale}/, "")
+    new_path = "/#{locale_string}#{current_path}"
```

Comment:
There is a better rails-y way to do it than with regular expressions:

```suggestion
    new_path = url_for(params.permit(:controller, :action, :site_id).merge(locale: locale_string))
```